### ISSUE TITLE
tpm_nvwrite dumping content

### ISSUE THREAD

**Issue by pohly** [intel/tpm2-tools#605]
When writing to NVRAM, I get a copy of the data in hex format on stdout:
```
# echo hello >/tmp/file
# TPM2TOOLS_TCTI_NAME=device tpm2_nvwrite -x 0x1500001 -a 0x40000001 -f /tmp/file
68 65 6c 6c 6f 0a 
```

That's probably not intentional, is it?

I'm using the 2.1.0 release.

**Comment by martinezjavier** [comment]
@pohly I agree with you that probably this information shouldn't be printed unconditionally. I think this is the kind of information that should be printed with LOG_INFO() only when the `--verbose` option is passed to the tool.

The problem is that the macro adds a newline character... I tried to change that in the past but @williamcroberts didn't like it.

**Comment by williamcroberts** [comment]
@poly the tools tend to spam users with all sorts of erroneous information. 

Newer versions of the tools on master and 3.X branch support -Q option and some/most of the tools respect that.

nvwrite still seems to be dumping to stdout the contents of the file:
```
        UINT16 i;
        for (i = 0; i < nv_write_data.t.size; i++) {
            nv_write_data.t.buffer[i] = ctx.nv_buffer[data_offset + i];
            tpm2_tool_output("%02x ", ctx.nv_buffer[data_offset + i]);
        }
        tpm2_tool_output("\n\n");
```

**Comment by williamcroberts** [comment]
-Q options gets rid of that. I could see it requiring -v, so that could ve blocked off in if flags.verbose check and skipped if verbose. I think this code should be using hexdump routine as well so its xxd compatible.

**Comment by williamcroberts** [comment]
Looks like tpm2_nvwrite is broken on master and 3.X as well.
```
$ ./tools/tpm2_nvwrite -x 0x1500015 -a 0x40000001 <<< 'foo'

$ ./tools/tpm2_nvread -x 0x1500015 -a 0x40000001
000000: 00 0b 6a a4 ff ff ff ff ff ff ff ff ff ff ff ff  ..j.............
```

What concerns me the most is that the nvread/write tests which I was running have cmp's on the data to find issues like this.

**Comment by martinezjavier** [comment]
Shouldn't you pass as an option the size `-s` to read? That's what the test do and probably why is working there.

**Comment by williamcroberts** [comment]
@martinezjavier 
If -s is omited it should grab it from the public nv info.
I read the data into nv_buffer and never use it again....

**Comment by williamcroberts** [comment]
closed with #617
